### PR TITLE
test: add comprehensive import type exclusion tests for #268

### LINKED ISSUE #268: IMPORTS-BUT-NO-CALLS: TypeScript import type statements inflate IBNC counts
## Anomaly

**Type**: IMPORTS-BUT-NO-CALLS
**Language**: TypeScript
**SBOM tools**: cdxgen v12.1.5
**Run**: 2026-04-11T1237 (diet-fuzz round 3)

## Summary

| Metric | Value |
|--------|-------|
| Affected projects | 1/3 (strapi) |
| Total anomalies | 1 (but 25 import files) |
| Most impacted | strapi/strapi (\`@strapi/types\`, 25 files, moderate difficulty) |

## Pattern

TypeScript's \`import type\` syntax is erased at compile time and produces no runtime code:

\`\`\`typescript
import type { Core, UID } from '@strapi/types';
\`\`\`

The tree-sitter import query matches all import statements:
\`\`\`
(import_statement source: (string) @import)
\`\`\`

This captures \`import type\` alongside regular imports. Since type-only imports have zero callable API surface at runtime, they always produce IBNC (import_file_count > 0, call_site_count = 0).

This is the TypeScript equivalent of Python's \`if TYPE_CHECKING:\` pattern — both are compile-time-only constructs.

### @strapi/types

| Dependency | Files | Calls | Score | Difficulty |
|-----------|-------|-------|-------|-----------|
| \`@strapi/types\` | 25 | **0** | 0.014 | **moderate** |

Examples from source:
\`\`\`typescript
// packages/core/review-workflows/server/src/register.ts
import type { Core, UID } from '@strapi/types';

// packages/core/review-workflows/server/src/destroy.ts
import type { Core } from '@strapi/types';

// packages/providers/upload-local/src/index.ts
import type {} from '@strapi/types';  // empty type import for module augmentation
\`\`\`

All 25 import files use \`import type\` exclusively. No runtime \`import\` exists.

## Suggested Fix

Modify the JS/TS import query to exclude type-only imports:

\`\`\`
;; Match regular imports but NOT import type
(import_statement
  !type            ;; exclude import_statement nodes with "type" flag
  source: (string) @import)
\`\`\`

Or, add a post-filter in \`extractImports()\` to check if the import statement has a \`type\` keyword. The exact tree-sitter node structure depends on the TypeScript grammar version.

## Prevalence

\`import type\` is very common in TypeScript codebases, especially:
- Monorepos with shared type packages (\`@strapi/types\`, \`@types/*\`)
- Libraries that separate types from implementation
- Projects using strict TypeScript with \`isolatedModules\`

Expect significant IBNC inflation in well-typed TypeScript projects.

## Reproduction

\`\`\`bash
git clone --depth 1 https://github.com/strapi/strapi.git /tmp/test
cdxgen -o /tmp/sbom.json /tmp/test
uzomuzo-diet --sbom /tmp/sbom.json --source /tmp/test --format json
# Check: @strapi/types → import_file_count=25, call_site_count=0
\`\`\`

---
*Filed by \`/diet-fuzz\` run 2026-04-11T1237*

### PR DESCRIPTION
**Pull request by kotakanbe**
## Summary
- Expand test coverage for TypeScript `import type` exclusion to cover all edge cases from issue #268
- The implementation (`isTypeOnlyImport` + `extractImports` check in `analyzer.go`) was already correct
- This PR adds missing regression test coverage for: scoped packages (strapi pattern), default type imports, empty type braces for module augmentation, mixed inline type specifiers, regular imports unaffected, and coexisting type-only/runtime imports in the same file

## Context
The `isTypeOnlyImport` function and its integration in `extractImports` (lines 366-371 of `analyzer.go`) were added during the treesitter refactoring. The previous test only covered the basic `import type { Foo, Bar } from "pkg"` case. This PR replaces that single-case test with a comprehensive table-driven test covering 7 edge cases identified in issue #268.

## Test output
```
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/import_type_with_named_imports_excluded
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/import_type_scoped_package_excluded_(strapi_pattern)
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/import_type_default_import_excluded
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/import_type_empty_braces_excluded
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/mixed_import_with_inline_type_specifier_still_counted
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/regular_named_import_unaffected
=== RUN   TestAnalyzer_TypeScriptImportTypeExclusion/type-only_and_regular_import_coexist_in_same_file
--- PASS: TestAnalyzer_TypeScriptImportTypeExclusion (0.31s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/import_type_with_named_imports_excluded (0.05s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/import_type_scoped_package_excluded_(strapi_pattern) (0.04s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/import_type_default_import_excluded (0.06s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/import_type_empty_braces_excluded (0.05s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/mixed_import_with_inline_type_specifier_still_counted (0.04s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/regular_named_import_unaffected (0.04s)
    --- PASS: TestAnalyzer_TypeScriptImportTypeExclusion/type-only_and_regular_import_coexist_in_same_file (0.03s)
PASS
```

Closes #268

## Test plan
- [x] Test `import type { X } from 'pkg'` excluded
- [x] Test `import type { Core, UID } from '@strapi/types'` excluded (exact strapi pattern)
- [x] Test `import type X from 'pkg'` (default type import) excluded
- [x] Test `import type {} from 'pkg'` (empty type import for module augmentation) excluded
- [x] Test mixed `import { type X, Y } from 'pkg'` still counted (has runtime binding Y)
- [x] Test regular `import { X } from 'pkg'` still counted
- [x] Test type-only and regular imports of different packages coexist correctly
- [x] go vet passes
- [x] golangci-lint passes
- [ ] CI green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by kotakanbe** [comment]
@claude Follow the procedure in `.github/prompts/review.prompt.md` with `--copilot-only` mode for this PR.

Phase 2: Resolve all unresolved Copilot review threads (discover → classify → fix → build/test/lint → commit/push → reply/resolve).
Phase 3: Detect recurring patterns from FIX-classified comments, propose and apply coding rules to `.github/instructions/`, then run `make sync-instructions`.
Phase 4: Post a summary comment on this PR.

**Comment by kotakanbe** [comment]
**Claude finished @kotakanbe's task in 2m 42s** —— [View job](https://github.com/future-architect/uzomuzo-oss/actions/runs/24288867015)

---
### Copilot-Only Review for PR #282

- [x] Phase 2: Discover and resolve Copilot review comments
  - [x] Read review prompt and test file
  - [x] Classify Copilot comment
  - [x] Fix the issue
  - [x] Build/test/lint
  - [x] Commit and push
  - [x] Reply and resolve thread
- [x] Phase 3: Detect recurring patterns, update coding rules
- [x] Phase 4: Post summary

---

### Copilot Resolution — PR #282

| # | File | Classification | Action |
|---|------|---------------|--------|
| 1 | `lang_javascript_test.go:866` | **FIX** | Added assertion that runtime imports produce coupling when coexisting with type-only imports |

Commit: 82b779d
Threads resolved: 1/1

### Rules Learned

| Pattern | Status | Detail |
|---------|--------|--------|
| testing | Accumulated | "Assert sibling items in multi-item wantNoResult tests" — PR #282 added to pending patterns |

Commit: 3c98efe

---
 ·

File: internal/infrastructure/treesitter/lang_javascript_test.go [hunk 1]
```diff
@@ -748,30 +748,158 @@ func TestAnalyzer_TypeScriptSideEffectImport(t *testing.T) {
...
+
+bar();
+`,
+			importPaths: map[string][]string{
+				"pkg:npm/type-only-pkg@1.0.0": {"type-only-pkg"},
+				"pkg:npm/runtime-pkg@1.0.0":   {"runtime-pkg"},
+			},
+			purlToCheck:  "pkg:npm/type-only-pkg@1.0.0",
+			wantNoResult: true,
+		},
```
[the remark this answers is not in the document]
Comment (kotakanbe) [reply internal/infrastructure/treesitter/lang_javascript_test.go:866]: Fixed in 82b779d.
